Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -18,15 +18,31 @@ import StoreKit
 @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
 class PurchasedProductsFetcher {
 
+    private typealias Transactions = [StoreKit.VerificationResult<StoreKit.Transaction>]
+
     private let appStoreSync: () async throws -> Void
     private let sandboxDetector: SandboxEnvironmentDetector
+    private let cache: InMemoryCachedObject<Transactions>
+    private let updatesObservation: Task<Void, Never>
 
     init(
         appStoreSync: @escaping () async throws -> Void = PurchasedProductsFetcher.defaultAppStoreSync,
         sandboxDetector: SandboxEnvironmentDetector = BundleSandboxEnvironmentDetector()
     ) {
         self.appStoreSync = appStoreSync
         self.sandboxDetector = sandboxDetector
+        self.cache = .init()
+
+        self.updatesObservation = Task<Void, Never>(priority: .utility) { [cache = self.cache] in
+            for await _ in StoreKit.Transaction.unfinished where cache.cachedInstance != nil {
```

Comment:
This isn't quite working.

Reply:
It's also making an integration test (I'm glad we have coverage for this there too):
```swift
@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
func testPurchasingTwoProductsWhileServerIsDown() async throws {
    let product1 = try await self.monthlyPackage.storeProduct
    let product2 = try await self.annualPackage.storeProduct

    self.serverDown()

    _ = try await Purchases.shared.purchase(product: product1)
    let info = try await Purchases.shared.purchase(product: product2).customerInfo

    try await self.verifyEntitlementWentThrough(info)
    expect(info.allPurchasedProductIdentifiers) == [
        product1.productIdentifier,
        product2.productIdentifier
    ]
}
```